tests/v2/validation/projects/rbac_terminating_project_test.go

```diff
@@ -0,0 +1,150 @@
+//go:build (validation || infra.any || cluster.any || extended) && !sanity && !stress
+
+package projects
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	projectsApi "github.com/rancher/shepherd/extensions/kubeapi/projects"
+	"github.com/rancher/shepherd/extensions/rancherleader"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RbacTerminatingProjectTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (pr *RbacTerminatingProjectTestSuite) TearDownSuite() {
+	pr.session.Cleanup()
+}
+
+func (pr *RbacTerminatingProjectTestSuite) SetupSuite() {
+	pr.session = session.NewSession()
+
+	client, err := rancher.NewClient("", pr.session)
+	require.NoError(pr.T(), err)
+
+	pr.client = client
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(pr.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(pr.client, clusterName)
+	require.NoError(pr.T(), err, "Error getting cluster ID")
+	pr.cluster, err = pr.client.Management.Cluster.ByID(clusterID)
+	require.NoError(pr.T(), err)
+
+}
+
+func (pr *RbacTerminatingProjectTestSuite) TestUserAdditionToClusterWithTerminatingProjectNamespace() {
+	subSession := pr.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a standard user.")
+	createdUser, err := users.CreateUserWithRole(pr.client, users.UserConfig(), projectsApi.StandardUser)
+	require.NoError(pr.T(), err)
+	pr.T().Logf("Created user: %v", createdUser.Username)
+
+	log.Info("Create a project in the downstream cluster.")
+	createdProject, err := createProject(pr.client, pr.cluster.ID)
+	require.NoError(pr.T(), err)
+	err = waitForFinalizerToUpdate(pr.client, createdProject.Name, createdProject.Namespace, 2)
```

Comment:
Correct me if I'm getting this wrong, do we want this step here?  because we are not adding a finalizer in the previous step where we are creating a project

Reply:
Good question. I have noticed intermittent issues where it takes a few seconds for the finalizers to get updated on project creation.  If we skip this step after project creation, subsequent step where we updateProjectNamespaceFinalizer may fail because the project has been updated, but we're still using outdated project information. Hope this makes sense

Reply:
oh, if I get it right, we are waiting for the finalizers that are automatically added on this project to be updated. correct?

F.e, I see these on one of my projects even when these were not updated by me manually:
  finalizers:
    - controller.cattle.io/mgmt-project-rbac-remove
    - clusterscoped.controller.cattle.io/project-namespace-auth_local

Reply:
That's correct, Anu! we are waiting for the finalizers that are automatically added on this project to be updated

Reply:
thanks for letting me know!